src/eth_provider/provider.rs

```diff
@@ -417,34 +418,31 @@ where
 
     async fn fee_history(
         &self,
-        mut block_count: U256,
+        block_count: U64HexOrNumber,
         newest_block: BlockNumberOrTag,
         _reward_percentiles: Option<Vec<f64>>,
     ) -> EthProviderResult<FeeHistory> {
-        if block_count == U256::ZERO {
+        if block_count.to() == 0 {
             return Ok(FeeHistory::default());
         }
 
-        let end_block = U256::from(self.tag_into_block_number(newest_block).await?);
-        let end_block_plus = end_block + U256::from(1);
+        let end_block = self.tag_into_block_number(newest_block).await?;
+        let end_block_plus = end_block + U64::from(1_u64);
 
         // Clamp the block count to the range [0, end_block_plus]
-        block_count = Ord::clamp(block_count, U256::ZERO, end_block_plus);
+        let block_count = Ord::clamp(U64::from(block_count.to()), U64::ZERO, end_block_plus);
```

Comment:
Should we have all variables be `u64`? It looks like we convert a lot to U64 but could probably just be u64 no?

Reply:
I just mean there is a lot of conversion between u64 and U64 in this function, maybe we can just use `u64` everywhere?